tests/integration/cmd_push_test.go

```diff
@@ -270,86 +163,10 @@ var _ = Describe("odo push command tests", func() {
 			Expect(output).To(Not(ContainSubstring("No file changes detected, skipping build")))
 		})
 
-		It("should push only the modified files", func() {
```

Comment:
this is a valid scenario that we should check.

Reply:
Tests that have been removed without a comment have a devfile counterpart already present hence can be deleted

Reply:
I was not able to find a test that would verify that `odo push` syncs only modified files. 
Would it make sense to convert this test to Devfile scenario?

Reply:
maybe

Reply:
I can do that

Reply:
Can we do this in a follow up https://github.com/openshift/odo/issues/4618

Reply:
you are removing the test case that tests one of the `odo push` primary functions :-/ 
I would prefer to have it in this PR. But I would say it is ok if you do it in follow up, but it needs to come right after this.

Reply:
> But I would say it is ok if you do it in follow up, but it needs to come right after this.

TBH I don't understand how does it make sense deleting the test in a pr then adding it in follow up pr. Is there a lot work involved?

Reply:
actually it does require some work but I am just trying to fix this PR in smaller chunks so that work moves along and next PRs will address smaller issues. This would also dependent team to use this change soon while we get feedback and I would get some breathing room 🙂

Reply:
done, added this function back and some helper functions that needed implementation

Reply:
https://github.com/openshift/odo/pull/4518/commits/ec833148b4082d0e9baca1e70711634ce1d4fe1d#diff-959d7173037ca057614f19d5dec825f383d808739d9d930933a6d6203405d203R156